features/crc/mbed_crc.h

```diff
@@ -0,0 +1,102 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_CRC_API_H
+#define MBED_CRC_API_H
+
+/** \addtogroup crc */
+/** @{*/
+#include <stdint.h>
+
+typedef uint64_t data_size_t;
```

Comment:
Should this be crc_data_size_t? or just crc_t?

data_size_t is ambiguous

Reply:
crc_data_size_t 👍 . Will change it